website/components/LandingPage/OpenSource/index.js

```diff
@@ -88,14 +90,27 @@ function dateRanges(from, to) {
 }
 
 function getNPMDownloads(from, to) {
+  function getRange(range, expires) {
+    return cached(
+      `npm-${range}`,
+      () =>
+        fetch(`https://api.npmjs.org/downloads/point/${range}/uniforms`)
+          .then(response => response.json())
+          .then(({ downloads }) => downloads || 0),
+      expires
+    );
+  }
+
   const dates = dateRanges(from, to);
+  const lastRange = dates.pop();
+
+  const oneYear = 12 * 31 * 24 * 60 * 60 * 1000;
+  const oneDay = 1 * 24 * 60 * 60 * 1000;
 
   return Promise.all(
-    dates.map(range =>
-      fetch(`https://api.npmjs.org/downloads/point/${range}/uniforms`)
-        .then(response => response.json())
-        .then(({ downloads }) => downloads || 0)
-    )
+    dates
+      .map(range => getRange(range, Date.now() + oneYear))
+      .concat(getRange(lastRange, Date.now() + oneDay))
```

Comment:
I'd rather use the date object methods instead of operating on milliseconds directly.

Reply:
In the end I still have to convert them to timestamp. This saves a lot of date object instances and conversions.

Reply:
OK, good.